### ISSUE TITLE
fireEvent() does not wait for rerender when imported as native ESM in Node v12

### ISSUE THREAD

**Issue by pastelmind** [testing-library/preact-testing-library#28]
- `preact-testing-library` version: 2.0.0
- `preact` version: 10.4.8
- `node` version: 12.18.3
- `npm` version: 6.14.8

Code that runs in Node.js v12 (saved as `app.mjs`):

```js
import { Component, h } from "preact";
import htm from "htm";
import TLP from "@testing-library/preact";
import jsdomGlobal from "jsdom-global";

const { fireEvent, render } = TLP;
const html = htm.bind(h);

class App extends Component {
  constructor() {
    super();
    this.state = { clickCount: 0 };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    console.log(
      `App.handleClick() called, clickCount == ${this.state.clickCount}`
    );
    this.setState((state) => ({ clickCount: state.clickCount + 1 }));
  }

  render() {
    console.log(`App.render() called, clickCount == ${this.state.clickCount}`);
    return html`
      <div>
        <button onClick="${this.handleClick}">
          Click count: ${this.state.clickCount}
        </button>
      </div>
    `;
  }
}

jsdomGlobal();

const { getByRole } = render(h(App));
const button = getByRole("button");

console.log("Before fireEvent");
fireEvent.click(button);
console.log("After fireEvent");
```

Equivalent code in browser: https://codesandbox.io/s/aged-pine-33292

**What you did:**

I am using Testing Library and Preact in Node.js, using JSDOM to simulate a browser environment. I ran `node --experimental-modules ./app.mjs`.

**What happened:**

When I run the script above in Node.js, the following log appears:

```
(node:4788) ExperimentalWarning: The ESM module loader is experimental.
App.render() called, clickCount == 0
Before fireEvent
App.handleClick() called, clickCount == 0
After fireEvent
App.render() called, clickCount == 1
```

Here, `fireEvent()` returned before the second `App.render()` call.

When I run the (almost) same code in browser:

```
App.render() called, clickCount == 0 
Before fireEvent
App.handleClick() called, clickCount == 0 
App.render() called, clickCount == 1 
After fireEvent
```

Here, `fireEvent()` returned _after_ the second `App.render()` call.

A curious discovery: `fireEvent()` behaves as expected under Node.js v10: https://codesandbox.io/s/frosty-blackwell-7686y . I assume that something is going haywire w.r.t. the [`"exports"`](https://nodejs.org/docs/latest-v12.x/api/esm.html#esm_package_entry_points) field in `package.json` of Preact--since Node.js v12 supports it, but Node.js v10 does not.

Some observations:

- Preact uses the `"exports"` field in `package.json` to export both ESM and CJS bundles.
- Preact Testing Library does not use the `"exports"` field, and therefore only exports CJS bundles.
- Only Node.js >= 12 can understand the `"exports"` field.
- Node.js uses separate loaders for ESM and CJS.
- When Node.js is running in native ESM mode, it can load both CJS and ESM modules.
    - ESM modules are loaded as-is
    - CJS modules are loaded through some compatibility magic provided by Node.js

My conjecture:

1. When I import Preact directly, Node.js v12 loads it through the ESM loader
    1. This causes Node.js to load the ESM bundle for Preact.
2. When I import Preact Testing Library, Node.js loads it through the CJS loader.
    1. Because of this, PTL's dependencies are also loaded through the CJS loader.
    2. Because of this, PTL loads the CJS bundle for Preact.
3. Node.js treats the ESM and CJS bundles of Preact as separate modules.
4. Because of this, `fireEvent()` uses `act()` from the _wrong_ copy of Preact, making it ineffective

Reproduction repository:

- Browser environment: https://codesandbox.io/s/aged-pine-33292
- Node.js v10 environment: https://codesandbox.io/s/frosty-blackwell-7686y
- I cannot provide a Node.js v12 environment at the moment, as CodeSandbox does not support them.

Problem description:

Suggested solution:

Perhaps Preact Testing Library could also provide an ESM bundle via the `"exports"` field? Since Node.js v12 is here to stay, we can expect the `"exports"` field to cause further headache unless it is addressed.

**Comment by pastelmind** [comment]
More evidence that this is an import problem: Wrapping `fireEvent()` inside an `act()` call causes the render to be flushed, _if `act()` is imported from `preact/test-utils`_:

```js
import { act } from "preact/test-utils";

// ...

// This works as expected
act(() => fireEvent.click(button));
```

...but not if `act()` is imported from Preact Testing Library:

```js
import TLP from "@testing-library/preact";
const { act } = TLP;

// ...

// This does not work in Node.js 12
act(() => fireEvent.click(button));
```

**Comment by aralroca** [comment]
Same problem. Any workaround?

**Comment by aralroca** [comment]
Replacing `fireEvent.change` to `fireEvent.input` solved my problem

**Comment by JoviDeCroock** [comment]
This could be fixed now with the addition of ESM as of 3.1.0
